src/data/padics/padic_norm.lean

```diff
@@ -112,6 +112,44 @@ lemma padic_val_rat_of_int (z : ℤ) (hp : p ≠ 1) (hz : z ≠ 0) :
     (finite_int_iff.2 ⟨hp, hz⟩) :=
 by rw [padic_val_rat, dif_pos]; simp *; refl
 
+/--
+A convenience function for the case of `padic_val_rat` when both inputs are natural numbers.
+-/
+def padic_val_nat (p : nat) (n : nat) : ℕ :=
+  int.to_nat (padic_val_rat p n)
+
+/--
+`padic_val_rat` is defined as an `int.to_nat` cast; this lemma ensures that the cast is well-behaved.
+-/
+lemma zero_le_padic_val_rat_of_nat (p n : nat) : 0 ≤ padic_val_rat p n :=
+begin
+  unfold padic_val_rat,
+  split_ifs,
+  { simp, },
+  { trivial, },
+end
+
+/--
+`padic_val_rat` coincides with `padic_val_nat`.
+-/
+lemma padic_val_rat_of_nat (p n : nat) : padic_val_rat p n = ↑(padic_val_nat p n) :=
```

Comment:
If you swap the order of the statement, then this would make a good `simp` and `norm_cast` lemma.